mlrun/projects/project.py

```diff
@@ -5266,6 +5266,12 @@ def _validate_file_path(self, file_path: str, param_name: str):
                 f"Invalid '{param_name}': '{file_path}'. Got a remote URL without a file suffix."
             )
 
+        _, in_context = self.get_item_absolute_path(file_path)
```

Comment:
This entirely can be removed. the validation should happen only if engine is provided and engine is NOT remote.
that means, if user provide `engine=remote` or `engine=remote:kfp` then you will skip the validation itself.